test/e2e/operator_test.go

```diff
@@ -100,51 +104,47 @@ func TestMain(m *testing.M) {
 		fmt.Printf("failed to get kube config: %s\n", err)
 		os.Exit(1)
 	}
-	cl, err := client.New(kubeConfig, client.Options{})
+	kubeClient, err = client.New(kubeConfig, client.Options{})
 	if err != nil {
 		fmt.Printf("failed to create kube client: %s\n", err)
 		os.Exit(1)
 	}
-	kubeClient = cl
-
-	if err := kubeClient.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, &infraConfig); err != nil {
-		fmt.Printf("failed to get infrastructure config: %v\n", err)
-		os.Exit(1)
-	}
-
 	kubeClientSet, err = kubernetes.NewForConfig(kubeConfig)
 	if err != nil {
 		fmt.Printf("failed to create kube clientset: %s\n", err)
 		os.Exit(1)
 	}
 
-	if !isOnROSA() {
-		if err := ensureCredentialsRequest(e2eSecretName); err != nil {
-			fmt.Printf("failed to create credentialsrequest for e2e: %s\n", err)
-			os.Exit(1)
-		}
-	} // ROSA forbids the creation of CredentialsRequests by non cluster admins.
-
 	var infra configv1.Infrastructure
-	clusterInfrastructureName := types.NamespacedName{Name: "cluster"}
-	err = kubeClient.Get(context.TODO(), clusterInfrastructureName, &infra)
+	err = kubeClient.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, &infra)
 	if err != nil {
-		fmt.Printf("failed to fetch infrastructure: %v", err)
+		fmt.Printf("failed to fetch infrastructure: %v\n", err)
 		os.Exit(1)
 	}
 
 	if infra.Status.PlatformStatus == nil || infra.Status.PlatformStatus.AWS == nil || infra.Status.PlatformStatus.AWS.Region == "" {
-		fmt.Printf("could not get AWS region from Infrastructure %q status", clusterInfrastructureName.Name)
+		fmt.Println("could not get AWS region from Infrastructure status")
 		os.Exit(1)
 	}
 
-	if !isOnROSA() {
+	if !stsModeRequested() {
+		if err := ensureCredentialsRequest(e2eSecretName); err != nil {
+			fmt.Printf("failed to create credentialsrequest for e2e: %s\n", err)
+			os.Exit(1)
+		}
 		cfg, err = awsConfigWithCredentials(context.TODO(), kubeClient, infra.Status.PlatformStatus.AWS.Region, e2eSecretName)
 		if err != nil {
-			fmt.Printf("failed to load aws config %v", err)
+			fmt.Printf("failed to load aws config %v\n", err)
 			os.Exit(1)
 		}
-	} // No need to get AWS config on ROSA cluster: the CI provisions all the required AWS resources.
+	} else {
+		// ROSA forbids the creation of CredentialsRequests by non cluster admins.
+		// TODO: provide IAM role for e2e test to unify the flows once ROSA can provision 4.14 clusters.
+		fmt.Println("Required AWS resources are expected to exist when the test is run on a ROSA STS cluster!")
```

Comment:
I tried to test this out on my STS cluster, and I failed here:
```
[gspence@gspence-thinkpadp1gen3 aws-load-balancer-operator]$ make ALBO_E2E_PLATFORM=OCPSTS ALBO_E2E_CONTROLLER_ROLE_ARN=$CONTROLLER_ROLEARN test-e2e 
go test \
-timeout 1h \
-count 1 \
-v \
-p 1 \
-tags e2e \
-run "" \
./test/e2e
Required AWS resources are expected to exist when the test is run on a ROSA STS cluster!
panic: environment variable "ALBO_E2E_WAFV2_WEBACL_ARN" must be set

goroutine 1 [running]:
github.com/openshift/aws-load-balancer-operator/test/e2e.mustGetEnv({0x191ff27, 0x19})
	/home/gspence/src/github.com/openshift/aws-load-balancer-operator/test/e2e/util.go:502 +0x8e
github.com/openshift/aws-load-balancer-operator/test/e2e.TestMain(0xffffffffffffffff?)
	/home/gspence/src/github.com/openshift/aws-load-balancer-operator/test/e2e/operator_test.go:144 +0x38d
main.main()
	_testmain.go:71 +0x1d3
FAIL	github.com/openshift/aws-load-balancer-operator/test/e2e	0.630s
FAIL

```

Reply:
I suppose if STS needs these WAF variables like ROSA, there isn't a problem. However, I do have a minor nit pick. In order to run a specific E2E test that doesn't need the WAF variables, I have to trick it by putting something in the variables:

`make ALBO_E2E_PLATFORM=OCPSTS ALBO_E2E_CONTROLLER_ROLE_ARN=$CONTROLLER_ROLEARN TEST=TestAWSLoadBalancerControllerWithDefaultIngressClass ALBO_E2E_WAFV2_WEBACL_ARN=test ALBO_E2E_WAF_WEBACL_ID=test test-e2e`

Does it make sense to limit these checks to when we use them? I'm don't feel too strongly, just pointing it out. I might be the only person ever trying this.

Reply:
> Do I still need these ENV variables for a non-ROSA STS clusters?

No.

> Does it make sense to limit these checks to when we use them?

Right, the WAF ACLs were checked in the concrete tests before for that reason.  Restoring this back to how it was for the WAF variables.